src/api/app/models/attrib_value.rb

```diff
@@ -1,48 +1,29 @@
-# This class represents a value inside of attribute part of package meta data
 class AttribValue < ApplicationRecord
-  #### Includes and extends
-  #### Constants
-  #### Self config
   acts_as_list scope: :attrib
-  after_initialize :init
-
-  #### Attributes
-  #### Associations macros (Belongs to, Has one, Has many)
   belongs_to :attrib
 
-  #### Callbacks macros: before_save, after_save, etc.
-  #### Scopes (first the default_scope macro if is used)
-  #### Validations macros
-  #### Class methods using self. (public and then private)
-  #### To define class methods as private use private_class_method
-  #### private
-  #### Instance methods (public and then protected/private)
-  def init
-    self.value ||= get_default_value
-  end
+  after_initialize :prepare_default_value
 
   def to_s
     value
   end
 
   private
 
-  # This defines the default for AttribValue.value to ""...
-  def get_default_value
-    value = ""
-    if read_attribute(:position).blank?
-      self.position = 1
-    end
+  def prepare_default_value
+    self.value = default_value if !value || value.blank?
```

Comment:
I think you can use ActiveRecord `attribute` class method to handle defaults. So it would be something like this:
```ruby
  attribute :position, :integer, default: 1
  attribute :value, :string, default: -> {
    if attrib
      default = attrib.attrib_type.default_values.find_by(position: position)
      default_value = default.value if default
    end

    default_value || ''
  }
```

Reply:
See: https://apidock.com/rails/ActiveRecord/Attributes/ClassMethods/attribute